### PR TITLE
[Merged by Bors] - Add support for Spark 3.4.0

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

For https://github.com/stackabletech/spark-k8s-operator/issues/242

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: tests/templates/kuttl/logging/05-deploy-automatic-log-config-spark-app.yaml.j2 [hunk 1]
```diff
@@ -5,13 +5,15 @@ metadata:
   name: spark-automatic-log-config
 spec:
   version: "1.0"
-  sparkImage: docker.stackable.tech/stackable/spark-k8s:{{ test_scenario['values']['spark'].split('-stackable')[0] }}-stackable{{ test_scenario['values']['spark'].split('-stackable')[1] }}
-  sparkImagePullPolicy: IfNotPresent
+  sparkImage: docker.stackable.tech/sandbox/sbernauer/spark-k8s:{{ test_scenario['values']['spark'].split('-stackable')[0] }}-stackable{{ test_scenario['values']['spark'].split('-stackable')[1] }}
```
Comment (sbernauer) [inline tests/templates/kuttl/logging/05-deploy-automatic-log-config-spark-app.yaml.j2:8]: Will revert it back to docker.stackable.tech/stackable/spark-k8s once images have been merged

**Review by maltesander — COMMENTED** [review]
LGTM overall. Will test a little now.

**Review by maltesander — CHANGES_REQUESTED** [review]
There are many references to `3.3.0-stackable0.3.0` instead e.g. `3.4.0-stackable0.0.0-dev` left over.

**Comment by sbernauer** [comment]
Started test https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/spark-k8s-operator-it-custom/86/

**Comment by sbernauer** [comment]
Bumped stackable versions.
Test passed as well :rocket:

**Review by maltesander — APPROVED** [review]
LGTM!

**Pull request closed without merging**